llvm/lib/MC/RepoObjectWriter.cpp

```diff
@@ -691,6 +699,15 @@ pstore::index::digest RepoObjectWriter::buildCompilationRecord(
       CompilationHash.update(Name.size());
       CompilationHash.update(stringViewAsRef(Name));
     }
+    // Update the fragment to remember the corrresponding compilation member.
+    if (FragmentPos != Fragments.end()) {
+      //  A reallocation shouldn't happen because the corresponding
+      //  compilation_member lies inside of Tickets and compilationMembers is
+      //  reserved to Tickets.size(),.
+      assert(CompilationMembers.capacity() == Tickets.size());
+      FragmentPos->second.CorrespondingCompilationMembersIts.push_back(
+          CompilationMembers.end() - 1);
```

Comment:
Sorry, missed this last time. Should there be an assertion that `CompilationMembers` is not empty? Instead of `CompilationMembers.end()-1` you can use `CompilationMembers.rbegin().base()`. (The "beginning" of a reverse iterator points to the last member of the container and `.base()` turns this into a conventional iterator.)

Reply:
I have updated the changes following your advice.